bcgov/HOUS-permit-portal
# Fix issue generating PDFs when deployed

## Description

Refactor the PDF generation code so it saves the PDF data to a temp file and passes the filename rather than the JSON directly to the node function. This fixes an error where the length of npm arguments is restricted.

Also addresses an issue with PDF filename specificity. The previous names are too generic as in the production environment, it is possible that multiple application PDFs are generated at the same timei. The name must be unique to the application to ensure the files are not overridden before the PDFs have finished generating.

## What type of PR is this? (check all applicable)


## Related Tickets & Documents

Implements [BPHH-1182](https://hous-bssb.atlassian.net/browse/BPHH-1182)

## Steps to QA

1. Rebuild node ssr: `npm run build:ssr`
2. Run PDF job to ensure PDFs are generated: `PdfGenerationJob.new.perform(permit_application_id)`

## UI Accessibility Checklist



_Note the following checklist is not an exhaustive list. Check
the [WebAIM checklist](https://webaim.org/standards/wcag/checklist) for
a user-friendly checklist with more details. Aim to meet at least *AA* conformance level where applicable. Check
the [WCAG guidelines](https://webaim.org/standards/wcag/checklist) for the full guidelines._

      with [Wave Browser Extension](https://chromewebstore.google.com/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh)
      for accessibility errors?

## General Checklist


## [optional] Are there any post-deployment tasks we need to perform?

n/a
